include/eve/module/core/named_shuffles/broadcast_lane.hpp

```diff
@@ -80,47 +80,47 @@ struct broadcast_lane_t
     else if constexpr( current_api >= vmx ) return 2;
     else if constexpr( current_api >= sve )
     {
-      if( !logical_value<T> ) return g_size > 8 ? 3 : 2;
-      if( G == 1 ) return 4;
-      if( g_size <= 8 ) return 6;
-      return 7;
+      if constexpr ( !logical_value<T> ) return g_size > 8 ? 3 : 2;
+      else if constexpr ( G == 1 ) return 4;
+      else if constexpr ( g_size <= 8 ) return 6;
+      else return 7;
     }
     else if constexpr( current_api >= neon )
     {
-      if( current_api >= asimd ) return 2;
-      if( reg_size <= 8 ) return 2;
-      return 4;
+      if constexpr ( current_api >= asimd ) return 2;
+      else if constexpr ( reg_size <= 8 ) return 2;
+      else return 4;
     }
 
     // x86
 
-    if (current_api == avx512 && logical_value<T>)
+    if constexpr (current_api == avx512 && logical_value<T>)
     {
-      if (G == 1) return 4;
-      return level(detail::mask_type(tgt), g, i) + 4;
+      if constexpr (G == 1) return 4;
+      else return level(detail::mask_type(tgt), g, i) + 4;
     }
-
-    if (reg_size == 64)
+    else if constexpr (reg_size == 64)
     {
-      if (g_size >= 16) return 2;
-      if (g_size >= 2) return 3;
-      return 4;
+      if constexpr (g_size >= 16) return 2;
+      else if constexpr (g_size >= 2) return 3;
+      else return 4;
     }
-
-    if (reg_size == 32)
+    else if constexpr (reg_size == 32)
     {
-      if (g_size >= 16) return 2;
-      if (current_api == avx) return 4;
-      if (g_size >= 8) return 2;
-      if (g_size >= 4) return 3;
-      if (g_size >= 2 && current_api == avx512) return 3;
-      return 4;
+      if constexpr (g_size >= 16) return 2;
+      if constexpr (current_api == avx) return 4;
+      if constexpr (g_size >= 8) return 2;
+      if constexpr (g_size >= 4) return 3;
+      if constexpr (g_size >= 2 && current_api == avx512) return 3;
```

Comment:
you see - now you push things to comile time switching instead of just evaluating. I don't know how this works under the hood, but it seems worse.